src/query/api/v1/httpd/handler.go

```diff
@@ -370,3 +430,34 @@ func (h *Handler) registerRoutesEndpoint() {
 		})
 	}).Methods(http.MethodGet)
 }
+
+func setupPrometheusMetricsHandler(listenAddress string) error {
```

Comment:
Seems like we do not use this method anywhere? We may need an optional address in config to expose the Prometheus metrics listen address on (if it's empty string we just do not expose the handler perhaps).

In future we want to merge the prometheus.Gatherer with our own gatherer and expose one single endpoint for scraping metrics from the process. 

We can tackle that it an later change however as it will be somewhat involved.